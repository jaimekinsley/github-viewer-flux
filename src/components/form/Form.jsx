import React from 'react';
import { useForm } from '../../hooks/appContext';
import { fetchUser } from '../../services/gitHubAPI';

const Form = () => {
  const { state, dispatch } = useForm();

  const handleChange = ({ target }) => {
    dispatch({
      type: 'SET_USERNAME',
      payload: target.value
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    fetchUser(state.username)
      .then(res => {
        dispatch({
          type: 'SET_USER_RESPONSE',
          payload: res
        });
      });
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={state.username} onChange={handleChange} />
      <button>Submit</button>
    </form>
  );
};

export default Form;
