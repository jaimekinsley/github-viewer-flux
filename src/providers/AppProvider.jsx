import React, { useReducer } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { useEffect } from 'react';
import { fetchUser } from '../services/gitHubAPI';

const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    fetchUser()
      .then(user => dispatch({ type: 'SET_USER', payload: user }));
  }, []);

  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppProvider;
