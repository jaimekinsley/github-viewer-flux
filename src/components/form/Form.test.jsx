import React from 'react';
import { mount } from 'enzyme';
import Form from './Form';
import AppProvider from '../../providers/AppProvider';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = mount(<AppProvider><Form /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
