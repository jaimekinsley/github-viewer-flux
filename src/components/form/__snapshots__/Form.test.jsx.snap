// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<AppProvider>
  <Form>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
      />
      <button>
        Submit
      </button>
    </form>
  </Form>
</AppProvider>
`;
